examples/3d/chess.rs

```diff
@@ -0,0 +1,482 @@
+            range: 100.,
+            ..Default::default()
+        },
+        ..Default::default()
+    });
+}
+
+fn create_board(
+    mut commands: Commands,
+    mut meshes: ResMut<Assets<Mesh>>,
+    mut materials: ResMut<Assets<StandardMaterial>>,
+) {
+    let mesh = meshes.add(Mesh::from(shape::Plane { size: 1.0 }));
+    let white_material = materials.add(Color::rgb(1.0, 0.9, 0.9).into());
+    let black_material = materials.add(Color::rgb(0.0, 0.1, 0.1).into());
+
+    // Add 64 Squares
+    for i in 0..8 {
+        for j in 0..8 {
+            commands.spawn_bundle(PbrBundle {
+                mesh: mesh.clone(),
+                material: if (i + j + 1) % 2 == 0 {
+                    white_material.clone()
+                } else {
+                    black_material.clone()
+                },
+                transform: Transform::from_translation(Vec3::new(
+                    i as f32 - 3.5,
+                    0.,
+                    j as f32 - 3.5,
+                )),
+                ..Default::default()
+            });
+        }
+    }
+}
+
+fn create_white_pieces(
+    mut commands: Commands,
+    asset_server: Res<AssetServer>,
+    mut materials: ResMut<Assets<StandardMaterial>>,
+) {
+    let king_handle: Handle<Mesh> =
+        asset_server.load("models/chess/pieces-all.glb#Mesh0/Primitive0");
+    let queen_handle: Handle<Mesh> =
+        asset_server.load("models/chess/pieces-all.glb#Mesh1/Primitive0");
+    let bishop_handle: Handle<Mesh> =
+        asset_server.load("models/chess/pieces-all.glb#Mesh2/Primitive0");
+    let knight_handle: Handle<Mesh> =
+        asset_server.load("models/chess/pieces-all.glb#Mesh3/Primitive0");
+    let rook_handle: Handle<Mesh> =
+        asset_server.load("models/chess/pieces-all.glb#Mesh4/Primitive0");
+    let pawn_handle: Handle<Mesh> =
+        asset_server.load("models/chess/pieces-all.glb#Mesh5/Primitive0");
+    let white_material = materials.add(Color::rgb(1.0, 0.8, 0.8).into());
+
+    commands
+        .spawn_bundle(PbrBundle {
+            transform: Transform::from_translation(Vec3::new(0.0, 0., 0.0)),
+            ..Default::default()
+        })
+        .with_children(|parent| {
+            parent.spawn_bundle(PbrBundle {
+                mesh: king_handle.clone(),
+                material: white_material.clone(),
+                transform: {
+                    let mut transform = Transform::from_translation(Vec3::new(-0.5, 0., 3.8));
+                    transform.apply_non_uniform_scale(Vec3::new(0.18, 0.18, 0.18));
+                    transform.rotate(Quat::from_rotation_x(1.65));
+                    transform.rotate(Quat::from_rotation_y(1.35));
+                    transform
+                },
+                ..Default::default()
+            });
+        })
+        .id();
+
+    commands
+        .spawn_bundle(PbrBundle {
+            transform: Transform::from_translation(Vec3::new(0., 0., 0.)),
+            ..Default::default()
```

Comment:
What's the reasoning for having empty parent pbr_bundles for every piece? It seems like the parent can be entirely omitted and the "child" instead spawned directly as the piece.

Reply:
The parent pieces follow a different coordinate system while the child piece local coordinate system is used to align the piece to the board. The grid for parent pieces is 0 -> 7, with the middle being 0 instead of 4. Meaning a rook on the left might move to +3.5x' from -3.5x. I'm still a little new to 3d programming; I'm open to doing this a better way!

Reply:
> The parent pieces follow a different coordinate system while the child piece local coordinate system is used to align the piece to the board. The grid for parent pieces is 0 -> 7, with the middle being 0 instead of 4. Meaning a rook on the left might move to +3.5x' from -3.5x. I'm still a little new to 3d programming; I'm open to doing this a better way!

I think I see what you're getting at but that's a bit confusing to me. The position of the "root" of an object should represent it's position in world space. As it is, the world position of all the pieces is at origin, with their children's local position offset to fit on the board.

To me the more intuitive way to do it would be to position each piece's parent transform to it's correct position/orientation on the board in world space. Then for example if a pawn needs to move forward by one, the parent transform for the pawn would move forward along it's local forward axis by one world unit. Does that make sense?